packages/web-app-files/src/views/Personal.vue

```diff
@@ -198,16 +204,80 @@ export default {
   },
 
   methods: {
-    ...mapActions('Files', ['loadIndicators', 'loadPreview']),
+    ...mapActions('Files', ['loadIndicators', 'loadPreview'], 'showMessage'),
     ...mapMutations('Files', [
       'SELECT_RESOURCES',
       'SET_CURRENT_FOLDER',
       'LOAD_FILES',
       'CLEAR_CURRENT_FILES_LIST',
-      'UPDATE_CURRENT_PAGE'
+      'UPDATE_CURRENT_PAGE',
+      'REMOVE_FILE_FROM_SEARCHED'
     ]),
     ...mapMutations(['SET_QUOTA']),
 
+    async fileDropped(fileIdTarget) {
+      const selected = this.selectedFiles
+      const targetInfo = this.getFileInfoById(fileIdTarget)
+      const targetSelected = selected.some(e => e.id === fileIdTarget)
+      if (targetSelected) return
+      if (targetInfo.type !== 'folder') return
+
+      this.loading = true
+      const targetPath = targetInfo.path + '/'
+      const errors = []
+      const itemsInTarget = await this.getFolderItems(targetPath)
+      for (let i = 0; i < selected.length; i++) {
+        const current = selected[i]
+        const exists = itemsInTarget.some(e => basename(e.name) === current.name)
+        if (exists) {
+          const message = this.$gettext('Resource with name %{name} already exists')
+          errors.push({
+            resource: current.name,
+            message: this.$gettextInterpolate(message, { name: current.name }, true)
+          })
+          continue
+        }
+        const promise = this.$client.files.move(current.path, targetPath + current.name)
+        await promise
+          .catch(error => {
+            error.resource = current.name
+            errors.push(error)
+          })
+          .then(() => {
+            this.REMOVE_FILE_FROM_SEARCHED(current)
+          })
+      }
+      this.loadResources(true)
+      this.loading = false
+      if (errors.length === 0) {
+        return
+      }
+      let title = this.$gettext('An error occurred while moving %{resource}')
+      if (errors.length === 1) {
```

Comment:
what if we have more then 1 error? Shouldn't i be just `if (errors.length)`

Reply:
its the same logic as in the locationpicker. If you look at the end of the "if" there is a return and after that the error handling for more than one error is addressed. It's ugly but we decided to leave it like this for now and make a refactoring task for extracting moving logic into a seperate file so we avoid code duplication.